docs/MANUAL.md

```diff
@@ -0,0 +1,215 @@
+# SonarDelphi Manual
+
+- [SonarDelphi Manual](#sonardelphi-manual)
+  - [SonarQube from first principles](#sonarqube-from-first-principles)
+  - [Configuring a SonarDelphi project](#configuring-a-sonardelphi-project)
+    - [Determining Sonar projects](#determining-sonar-projects)
+    - [Configuring Sonar projects](#configuring-sonar-projects)
+    - [Advanced project configuration](#advanced-project-configuration)
+  - [Run a SonarDelphi scan](#run-a-sonardelphi-scan)
+  - [Configure analysis rules](#configure-analysis-rules)
+    - [Customize analysis using template rules](#customize-analysis-using-template-rules)
+    - [Add new custom rules programmatically](#add-new-custom-rules-programmatically)
+
+## SonarQube from first principles
+
+SonarDelphi is a plugin that adds Delphi language support to the Sonar code analysis platform.
+
+Sonar has two main components:
+
+* A **server** (SonarQube):
+  * Stores project scan results ("issues" and "metrics")
+  * Has a web UI that can be used to browse issues detected in source files
+  * Allows users to change issue statuses, add comments to issues, etc.
+  * Supports various plugins that provide functionality for the server, the scanner, or both
+* A **scanner** "scans" a project locally by downloading plugins from the server and running them on the local files
+  * Each plugin:
+    1. Reads the source code files that are relevant to it
+    2. Calculates **metrics** (e.g. lines of code, coverage, etc.)
+    3. Analyzes the source code, detecting **issues** by applying **rules**
+  * At the end of a scan, the scanner sends the metrics and issues to the server
+
+This is a very high-level description - for more details, please refer to the
+[SonarQube documentation](https://docs.sonarsource.com/sonarqube/latest/).
+
+As a plugin, SonarDelphi's job is to:
+
+1. Read and understand Delphi code,
+2. Calculate metrics in Delphi code, and
+3. Detect issues in Delphi code.
+
+## Configuring a SonarDelphi project
+
+### Determining Sonar projects
```

Comment:
Not loving the term "determining" or "dividing" to describe this process, but I'm not sure of a good (and terse) way to describe it...

Reply:
I use "mapping" in other places?

Reply:
Replaced with "scoping".